bin/circleci-run

```diff
@@ -1,25 +1,52 @@
 #!/usr/bin/env ruby
 
-##
-# There are lots of moving parts so the tests can be flaky.
-# For better or worse we only sound the alarm if we fail after 3 tries.
-#
-def run_and_retry_tests(environment)
-  result = system "HEADLESS_BROWSER=true LOWER_ENV=#{environment} bundle exec rspec"
-  return if result
-
-  2.times do
+require 'optparse'
+
+def run_and_retry_tests(environment, retries)
+  extra_flags = ''
+  retries.times do
+    return if system "HEADLESS_BROWSER=true LOWER_ENV=#{environment} bundle exec rspec#{extra_flags}"
+
     puts "Smoke tests failed in #{environment}. Retrying..."
-    return if system "HEADLESS_BROWSER=true LOWER_ENV=#{environment} bundle exec rspec --only-failure"
+    extra_flags = ' --only-failure'
```

Comment:
WDYT of making the command an array and joining it, rather than having to manage spaces manually?

```ruby
cmd = ['HEADLESS_BROWSER=true LOWER_ENV=#{environment} bundle exec rspec']
cmd << '--only-failure'
# ...
system cmd.join(' ')
```